src/test/java/com/redhat/devtools/lsp4ij/features/foldingRange/JavaFoldingRangeTest.java

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.foldingRange;
+
+import com.intellij.codeInsight.folding.CodeFoldingSettings;
+import com.redhat.devtools.lsp4ij.fixtures.LSPFoldingRangeFixtureTestCase;
+
+/**
+ * Folding range tests by emulating LSP 'textDocument/foldingRange' responses from jdtls.
+ */
+public class JavaFoldingRangeTest extends LSPFoldingRangeFixtureTestCase {
```

Comment:
I added a test for Java, but it's not as comprehensive as the one for TypeScript as it would otherwise just be almost exactly the same. I did include tests specifically for its support for both `comment` and `imports` folding range kinds, though, and I confirm that regions are collapsed-by-default as appropriate based on the respective IDE configuration settings.